detectors/node/opentelemetry-resource-detector-docker/src/detectors/DockerCGroupV1Detector.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  Detector,
+  Resource,
+  ResourceDetectionConfig,
+} from '@opentelemetry/resources';
+
+import { SemanticResourceAttributes } from '@opentelemetry/semantic-conventions';
+
+import * as fs from 'fs';
+import * as util from 'util';
+import { diag } from '@opentelemetry/api';
+
+export class DockerCGroupV1Detector implements Detector {
+  readonly CONTAINER_ID_LENGTH = 64;
+  readonly DEFAULT_CGROUP_PATH = '/proc/self/cgroup';
+  readonly TIMEOUT_MS = 2000;
+  readonly UTF8_UNICODE = 'utf8';
+
+  private static readFileAsync = util.promisify(fs.readFile);
+  private static fileAccessAsync = util.promisify(fs.access);
+
+  async detect(_config?: ResourceDetectionConfig): Promise<Resource> {
+    try {
+      if (!(await this._isSupportedDocker())) {
```

Comment:
[fs docs](https://nodejs.org/api/fs.html#fspromisesaccesspath-mode) seem to agree:

> Using fsPromises.access() to check for the accessibility of a file before calling fsPromises.open() is not recommended. Doing so introduces a race condition, since other processes may change the file's state between the two calls. Instead, user code should open/read/write the file directly and handle the error raised if the file is not accessible.

Reply:
@Rauno56 That is a good insight , thank you for the comment. I will update the PR and notify .

Reply:
So - I will remove the check itself and if there is any issue - it should be caught in the catch. Right ?

Reply:
Yes - The catch surrounding `readFile`.

Reply:
done